framework/src/utils/ElementsIntersectedByPlane.C

```diff
@@ -0,0 +1,79 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+// Moose includes
+#include "ElementsIntersectedByPlane.h"
+
+// libMesh includes
+#include "libmesh/plane.h"
+#include "libmesh/point.h"
+#include "libmesh/mesh.h"
+#include "libmesh/elem.h"
+
+#include <algorithm>
+
+namespace Moose
+{
+
+void findElementsIntersectedByPlane(const Plane & plane, const MeshBase & mesh, std::vector<Elem *> & intersected_elems)
+{
+  // Loop over all elements to find elements intersected by the plane
+  MeshBase::const_element_iterator el = mesh.elements_begin();
+  const MeshBase::const_element_iterator end_el = mesh.elements_end();
+  for (; el != end_el; ++el)
+  {
+    const Elem * elem = *el;
+    const unsigned int dim = elem->dim();
+    std::vector<bool> above_plane;
+
+    // Loop over all nodes belonging to this element and check if they are above the plane or not
+    for (unsigned int i = 0; i < elem->n_nodes(); ++i)
+    {
+      const Node node = *elem->get_node(i);
+
+      bool node_above = plane.above_surface(node);
+      above_plane.push_back(node_above);
```

Comment:
Suggestion: This might cause multiple reallocations. You might size the vector by `n_nodes()` above. Alternatively you might come up with a more clever way of detecting two values. You don't need to save them all, you just need to know when you've seen two different values. Things like `xor` or maybe assign true as the 0x01 bit and false as 0x02 in an unsigned char and seeing if at the end both bits are set. All optimizations though so not necessary for now.